src/storage/v2/vertex_accessor.cpp

```diff
@@ -706,15 +734,39 @@ Result<EdgesVertexAccessorResult> VertexAccessor::OutEdges(View view, const std:
   {
     auto guard = std::shared_lock{vertex_->lock};
     deleted = vertex_->deleted;
-    expanded_count = static_cast<int64_t>(vertex_->out_edges.size());
-    if (edge_types.empty() && !destination) {
-      out_edges = vertex_->out_edges;
+    if (hops_limit && hops_limit->IsUsed()) {
+      if (edge_types.empty() && !destination) {
+        // we need this check because if limit is reached but it is not passed we can reach use vertices without any
+        // neighbors
+        if (hops_limit->LeftHops() == 0 && static_cast<int64_t>(vertex_->out_edges.size()) > 0) {
+          hops_limit->limit_passed = true;
```

Comment:
Why do we need this flag? I couldn't find a place where do we read it. We only set values

Reply:
limit_reached flag should be used, no need for additional flag.